doc/release-notes/7440-multilicense.md

```diff
@@ -0,0 +1,119 @@
+
+
+## Major Use Cases and Infrastructure Enhancements
+
+
+- When creating/updating datasets, users can select from a set of standard licenses configured by the administrator or provide custom terms (if the installation is configured to allow them).
+
+## Notes for Dataverse Installation Administrators
+
+### Updating for multiple license support
+
+As part of installing/upgrading an existing installation, administrators may wish to add additional license choices and/or configure Dataverse to allow custom terms. Adding additional licenses is managed via API. Licenses are described via a JSON structure providing a name, URL, short description, and optional icon URL. Additionally licenses may be marked as active (selectable for new/updated datasets) or inactive (only allowed on existing datasets) and one license can be marked as the default. Custom Terms are allowed by default (backward compatible with the current option to select 'No' to using CC0) and can be disabled by setting `:AllowCustomTermsOfUse` to false.
+
+Further administrators should review the following automated migration of existing licenses and terms into the new license framework and, if desired, should manually find and update any datasets for which the automated update is problematic. 
+To understand the migration process, it is useful to understand how the multiple license feature works in this release:
+
+'Custom Terms', aka a custom license, are defined through entries in the following fields of the dataset "Terms" tab:
+- Terms of Use
+- Confidentiality Declaration 
+- Special Permissions 
+- Restrictions 
+- Citation Requirements 
+- Depositor Requirements 
+- Conditions 
+- Disclaimer
+
+'Custom Terms' require,at a minimum, a non-blank entry in the "Terms of Use" field. Entries in other fields are optional.
+
+Since these fields are intended for terms/conditions that would potentially conflict with or modify the terms in a standard license, they are no longer shown when a standard license is selected.
+
+In earlier Dataverse releases, it was possible to select the CC0 license and have entries in the fields above. It was also possible to say 'No' to using CC0 and leave all of these terms fields blank. 
+
+The automated process will update existing datasets as follows.
+
+- 'CC0 Waiver' and no entries in the fields above -> CC0 License (no change)
+- No CC0 Waiver and an entry in the Terms of Use field and possibly others fields listed above -> 'Custom Terms' with the same entries in these fields (no change)
+
+- CC0 Waiver and an entry in some of the fields listed -> 'Custom Terms' with the following text preprended in the "Terms of Use" field: "This dataset is made available under a Creative Commons CC0 license with the following additional/modified terms and conditions:"
```

Comment:
Yes. I'm not sure that anyone will think that is a good default, but we added something just so that a naïve update won't lose info or result in an invalid/legacy case where a dataset now has a CC0 license and custom terms - something you can't create with the new UI. 

(Not sure if it has been said explicitly anywhere yet but in the design and review of this feature, the idea that one should not be able to add terms of use (the specific termsOfUse field  and others that also present a way to modify/counter license terms) was agreed to, as was the idea that having a custom license and a blank/empty terms of use field (essentially having no terms at all) should no longer be allowed. These have created the question about how to handle legacy datasets that violate those restrictions. From looking at Harvard and some other sites, it seems like these cases should be rare, so the idea of providing queries/a process for finding and fixing them prior to upgrading (could be done way before upgrading too) and then providing some default that would catch any remaining cases in a way that works technically and arguably doesn't change the meaning, seems like an acceptable approach. FWIW: to help clarify things for the community, I was also hoping to do a recorded segment in a community meeting that would show off the new multi-license feature and explain what would happen to existing data/how to avoid the default upgrades if desired.))

Reply:
Sure, I definitely think showing off the new feature during a community call would be interesting and helpful.

Makes sense what you're saying about converting invalid/legacy cases to valid ones.